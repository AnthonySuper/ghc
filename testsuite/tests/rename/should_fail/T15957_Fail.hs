{-# LANGUAGE RecordWildCards #-}
{-# LANGUAGE NamedFieldPuns #-}
module T15957_Fail where

data P = P { x :: Int, y :: Int }

f1 P{..} = 1 + 3 -- nothing bound is used
f2 P{x, ..} = x + 3 -- y bound but not used
f3 P{x, y, ..} = x + y -- no bindings left, i.e. no new useful bindings introduced


