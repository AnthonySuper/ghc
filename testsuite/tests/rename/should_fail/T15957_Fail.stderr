
T15957_Fail.hs:7:4: error: [-Wunused-record-wildcards, -Werror=unused-record-wildcards]
    No variables bound in the record wildcard match are used

T15957_Fail.hs:8:4: error: [-Wunused-record-wildcards, -Werror=unused-record-wildcards]
    No variables bound in the record wildcard match are used

T15957_Fail.hs:9:4: error: [-Wredundant-record-wildcards, -Werror=redundant-record-wildcards]
    Record wildcard does not bind any new variables
